src/web-pathing.md

```diff
@@ -0,0 +1,143 @@
+  - TODO `/ipfs/unixfs-dir-cid/dag-cbor-file/cbor-field`  (boxo/gateway errors on  this ([spec→traversing-cbor notes](https://specs.ipfs.tech/http-gateways/path-gateway/#traversing-through-dag-json-and-dag-cbor)), but we should specify behavior when someone wants to support this)
+  - TODO make it clear if both DAG variants of CBOR and JSON are a MUST, or if JSON is a SHOULD (right now conformance tests require both as a MUST).
+
+- TODO: MUST what happens when we can't traverse part of the path
+  - TODO: separate errors for traversal errors due to missing codec vs missing content
+    - TODO: `/ipfs/valid-cid-dag-pb/invalid-path` (logical "not found", translates to HTTP 404 to indicate content does not exist, mention implicit http caching of 404 vs 500 – )
+    - TODO: `/ipfs/cid/unknown-codec-block/some/path` is requested (logical "path parser error", translates to  HTTP 500 error page due to missing decoder)
+
+- TODO: MUST describe handling of non-ascii characters
+  - TODO: dont invent anything new, refer to URL percent-encoding, like we did in [IPIP-383](https://github.com/ipfs/specs/pull/383)
+    - TODO: non-ascii characters (percent-encoding of unicode and arbitrary binary data)
+    - TODO: MUST: explicitly cover Unicode and that UTF-8 is implicit default
+    - TODO: have an answer for non-UTF-8 (e.g. UTF-16) code points (a MAY and error if are not supported? or error since this is web pathing, and web URL encoding uses UTF-8?)
+    - TODO: edge case: handling filenames that already look percent-encoded https://github.com/ipfs/gateway-conformance/issues/115
+      - TODO/TBD notes for implementers: mixing percent-encoded and raw paths is a very very comon case across the stack, writing down a sane MUST rule of thumb for implementers could improve resiliency across systems (e.g. if path includes `%` and produced 404, retry with percent-decoded value?)
+
+- TODO: path normalization
+  - TODO: note that paths are equivalent, but HTTP 301 SHOULD be used in HTTP context to ensure clients always end up on normalized paths
+  - TODO: handling redundant slashes  `///`  (301 to resolved URL? `path.Clean`?)
+  - TODO: handling `.` and `..` (301 to resolved URL? `path.Clean`?)
+  - TODO: trailing slash `/` required for enumerable map-like entities (UnixFS dir, DAG-CBOR document?)
+  - TODO: CID normalization (to canonical text respresentation version and multibase)
+    - /ipfs to CIDv1 in base32
+    - /ipns to CIDV1 with libp2p-key in base36
+
+### Test fixtures
+
+TODO: List relevant CIDs. Describe how implementations can use them to determine
+specification compliance.
+
+TODO: [gateway-conformance](https://github.com/ipfs/gateway-conformance) tests for all MUSTs in this spec
+This ensure uniform behavior across implementations and contexts such as gateways vs `ipfs://` in browsers
+
+### Security
+
+TODO: Explain the security implications/considerations relevant to the spec.
+
+TODO: length limit for entire path
```

Comment:
Are we limiting to browser URLs, or do we want to support longer lengths? https://stackoverflow.com/a/417184/592760 is a really thorough answer talking about variants.